code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -329,21 +329,37 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 	execute(var/obj/item/owner, var/mob/attacker, var/mob/attacked)
 		if(prob(50))
 			if(istype(attacked) && !isrestrictedz(attacked.z)) // Haine fix for undefined proc or verb /turf/simulated/floor/set loc()
-				. = get_turf(pick(orange(8, attacked)))
-				if (isturf(.))
-					attacked.visible_message("<span class='alert'>[attacked] is warped away!</span>")
-					boutput(attacked, "<span class='alert'>You suddenly teleport ...</span>")
-					attacked.set_loc(.)
+				if(istype(attacked.loc, /obj/machinery/vehicle/)) // Lord knows how you'd manage to get attacked while *inside* a vehicle, but better safe than sorry.
+					var/obj/machinery/vehicle/V = attacked.loc
+					. = get_turf(pick(orange(8, V)))
+					if (isturf(.))
+						attacked.visible_message("<span class='alert'>[attacked] is warped away!</span>")
+						boutput(attacked, "<span class='alert'>You suddenly teleport ...</span>")
+						attacked.set_loc(.)
+				else
+					. = get_turf(pick(orange(8, attacked)))
+					if (isturf(.))
+						attacked.visible_message("<span class='alert'>[attacked] is warped away!</span>")
+						boutput(attacked, "<span class='alert'>You suddenly teleport ...</span>")
+						attacked.set_loc(.)
```

Comment:
Wouldn't just having the original code with something like `. = get_turf(pick(orange(8, get_turf(attacked))))` work or am I missing something?
(Also as a sidenote doing this through `pick(orange( ))` is terribly inefficient because it has to generate the complete list of all objects 8 or less tiles away from the location.)

Reply:
Faster way would be to get x, y, z and then add a rand(-8, 8) to it. (Or ideally make a helper proc for that kind of thing if it doesn't exist already.)

Reply:
Well, now that I know about get_turf (I didn't when I wrote that) that'd probably work?

Where would you suggest I put that proc if I don't find one? I'm not familiar enough with the repo yet to know where general-purpose procs go.

Reply:
There's something like helperprocs.dm for unsorted utility procs like that.

Reply:
`code\procs\helpers.dm:908`, `/proc/get_offset_target_turf(var/atom/A, var/dx, var/dy)`
e.g.: `. = get_offset_target_turf(get_turf(attacked), rand(-4, 4), rand(-4, 4))`